bibliopixel/remote/control.py

```diff
@@ -14,89 +13,95 @@
     'valid': True
 }
 
-DEFAULT_SERVER_CONFIG = {
-    'title': 'BiblioPixel Remote',
-    'bgcolor': '#000000',
-    'font_color': '#ffffff',
-    'external_access': False,
-    'port': 5000
+
+OFF_ANIM = {
+    'animation': 'bibliopixel.animation.off.OffAnim',
+    'run': {'threaded': True}
 }
 
+
 BAD_DEFAULT_ERROR = """\
 `{}` is not a valid default!
 It must be one of the configured animation names."""
 
 
-class RemoteControl:
-    def __init__(self, config, animations):
-        # These start with the defaults
-        self.config = dict(DEFAULT_SERVER_CONFIG, **config)
+class RemoteControl(collection.Collection):
+    def __init__(self, layout, animations,
+                 external_access=False, port=5000,
+                 title='BiblioPixel Remote', bgcolor='black',
+                 font_color='white', default=None):
+
+        self.internal_delay = 0  # never wait
+
+        # normalize name and display first
+        for anim in animations:
+            anim['animation']['data'] = dict(DEFAULT_ANIM_CONFIG, **anim['animation'].get('data', {}))
+            if not anim['animation'].get('name'):
+                raise ValueError('All animations require the `name` parameter: {}'.format(anim['animation']))
+            if not anim['animation']['data'].get('display'):
+                anim['animation']['data']['display'] = anim['animation']['name']
+            anim['animation']['name'] = ''.join(e for e in anim['animation']['name'] if e.isalnum())
+            anim['run'] = anim.get('run', {})
+            anim['run']['threaded'] = True  # threaded required
+        super().__init__(layout, animations, True)
+
+        self.name_map = {}
+        self.anim_cfgs = []
+        for i in range(len(self.animations)):
+            anim = self.animations[i]
+            name = animations[i]['animation']['name']  # if failed to load anim will be None
```

Comment:
at this point `animations[i]` is the dict config and `self.animations[i]` is the actual class object.
The reason I did this is that if the animation fails to load, then `self.animations[i]` is None which will have no `name` property. I though about instead of returning None returning like a BaseAnimation instance but that just seemed hackish. But I admit, this is not clear. The BaseAnimation thing would work but I would need to somehow check that that's all it is, therefore it didn't actually load. Thoughts?